x-pack/legacy/plugins/apm/public/components/app/Settings/AgentConfigurations/AgentConfigurationCreateEdit/SettingsPage/SettingsPage.tsx

```diff
@@ -0,0 +1,233 @@
+  EuiStat,
+  EuiBottomBar,
+  EuiText,
+  EuiHealth
+} from '@elastic/eui';
+import React, { useState, useMemo } from 'react';
+import { i18n } from '@kbn/i18n';
+import { AgentName } from '../../../../../../../../../../plugins/apm/typings/es_schemas/ui/fields/agent';
+import { history } from '../../../../../../utils/history';
+import { AgentConfigurationIntake } from '../../../../../../../../../../plugins/apm/common/runtime_types/agent_configuration/configuration_types';
+import {
+  filterByAgent,
+  configSettingDefinitions,
+  isValid
+} from '../../../../../../../../../../plugins/apm/common/runtime_types/agent_configuration/config_setting_definitions';
+import { saveConfig } from './saveConfig';
+import { useApmPluginContext } from '../../../../../../hooks/useApmPluginContext';
+import { useUiTracker } from '../../../../../../../../../../plugins/observability/public';
+import { SettingFormRow } from './SettingFormRow';
+import { getOptionLabel } from '../../../../../../../../../../plugins/apm/common/agent_configuration_constants';
+import { CancelButton } from '../ServicePage/CancelButton';
+
+export function SettingsPage({
+  unsavedChanges,
+  newConfig,
+  setNewConfig,
+  isEditMode,
+  onClickEdit
+}: {
+  unsavedChanges: Record<string, string>;
+  newConfig: AgentConfigurationIntake;
+  setNewConfig: React.Dispatch<React.SetStateAction<AgentConfigurationIntake>>;
+  isEditMode: boolean;
+  onClickEdit: () => void;
+}) {
+  // get a telemetry UI event tracker
+  const trackApmEvent = useUiTracker({ app: 'apm' });
+  const { toasts } = useApmPluginContext().core.notifications;
+  const [isSaving, setIsSaving] = useState(false);
+  const unsavedChangesCount = Object.keys(unsavedChanges).length;
+
+  const isFormValid = useMemo(() => {
+    return (
+      configSettingDefinitions
+        // only validate settings that are not empty
+        .filter(({ key }) => {
+          const value = newConfig.settings[key];
+          return value != null && value !== '';
+        })
+
+        // every setting must be valid for the form to be valid
+        .every(def => {
+          const value = newConfig.settings[def.key];
+          return isValid(def, value);
+        })
+    );
+  }, [newConfig.settings]);
+
+  const handleSubmitEvent = async () => {
+    trackApmEvent({ metric: 'save_agent_configuration' });
+    const config = { ...newConfig, settings: removeEmpty(newConfig.settings) };
+
+    setIsSaving(true);
+    await saveConfig({ config, isEditMode, toasts });
+    setIsSaving(false);
+
+    // go back to overview
+    history.push({
+      pathname: '/settings/agent-configuration',
+      search: history.location.search
+    });
+  };
+
+  return (
+    <>
+      <EuiForm>
+        {/* Since the submit button is placed outside the form we cannot use `onSubmit` and have to use `onKeyPress` to submit the form on enter */}
+        {/* eslint-disable-next-line jsx-a11y/no-noninteractive-element-interactions */}
+        <form
```

Comment:
Can't you place the form outside? in the same scope as the button?

Reply:
The submit button is moved to another place in the DOM with portals so I don't think it's possible.